### PR TITLE
Issue #287 - Introduce Tycho Pomless Build and Maven Wrapper

### PR DESCRIPTION
**Pull request by Boereck**
This change gets rid of most pom.xml files in Corrosion. This is done
by using the Tycho Pomless technology in it's latest version. Since
Tycho Pomless is somehow sensitive to different maven versions, the
maven wrapper technology is used. This allows users (even without a
maven installation) to always use the recommended maven version to build
Corrosion. Build instructions were updated accordingly.

The update-mvnw scripts were added to show how to update the maven wrapper 
or the maven version used by the wrapper.

The maven wrapper technology was approved via CQ 21342:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=21342

The added rustup-init.sh file was previously downloaded as part of the
build and was already approved via CQ 21017:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=21017

Signed-off-by: Max Bureck <max.bureck@fokus.fraunhofer.de>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/mvnw
+/mvnw.cmd
+/.mvn/wrapper/*
```
Comment (mickaelistria) [inline .gitignore]: I see all those are tracked by git, shouldn't this gitignore be removed?

Comment (Boereck) [reply .gitignore]: Yes! This is a leftover of a previous iteration. Thanks!

**Comment by Boereck** [comment]
`mvnw` was not executable, so the Jenkins build failed. Let's see if it works now.

**Comment by Boereck** [comment]
Build was still failing. Probably needed to call `./mvnw` instead of `mvnw` in Jenkins file.

**Pull request merged**